### ISSUE TITLE
Scenario Outlines with example params in title break report formatting

### ISSUE THREAD

**Issue by eitzenbe** [serenity-bdd/serenity-core#3522]
### What happened?

Cloned serenity-cucumber-starter project, upgraded to serenity version 4.1.20 (but also fails with original version)

changed feature file to:

Feature: Search by keyword

  @green
  Scenario: Searching for 'green'
    Given Sergey is researching things on the internet
    When he looks up "green"
    Then he should see information about "green"

  @red
  Scenario Outline: Searching for "<color>"
    Given Sergey is researching things on the internet
    When he looks up "<color>"
    Then he should see information about "<color>"
    Examples:
    |color|
    |blue |

Report looks strange:

![image](https://github.com/user-attachments/assets/1b6d35c8-4f88-4685-9b9c-ec42b9f506eb)

First entry is from a version where the title was without "<color" but only <color> in title

![image](https://github.com/user-attachments/assets/8b83c590-079b-4f99-be00-ab15d39ee471)

Side Note: This happens only on testresults pages, on the feature pages its all fine:

![image](https://github.com/user-attachments/assets/ab2c8f32-bbae-4d15-b69f-c788cf4689e0)

![image](https://github.com/user-attachments/assets/c2c7735b-e2c3-4c37-b893-87c6d96326b7)

### What did you expect to happen?

The Testresults pages/tables show the title with correctly escaped "<" and ">" tags

### Serenity BDD version

4.1.20

### JDK version

jdk17

### Execution environment

Linux

### How to reproduce the bug.

see above, minimal changes to be applied to serenity-cucumber-starter project

### How can we make it happen?

Work on this myself and propose a PR (with Serenity BDD team guidance)

**Comment by wakaleo** [comment]
Well spotted - the thing to do would be to check the reporting logic for the feature files (start with the freemarker templates) and see what text transformation functions are used there. You should be able to just replicate these in the other template.

**Comment by eitzenbe** [comment]
any filename or folder to start with? ;)

**Comment by eitzenbe** [comment]
mvn install -DskipTests running ;)

**Comment by wakaleo** [comment]
Yep, you can test a snapshot build that way (don't run the full test suite each time, as the full test suite takes a while and is a bit Linux-centric for some of the tests).

**Comment by eitzenbe** [comment]
I am off course on linux ;)

**Comment by eitzenbe** [comment]
https://github.com/serenity-bdd/serenity-core/pull/3523  avail for review, I did a minimal invasive fix, which might not be the most elegant way, but I don't want to open up pandoras box  here ;) I tested with scenario titles containing <,>," as these would fuck the report HTML the most and it seems you are checking for a lot of special chars but not these

**Comment by eitzenbe** [comment]
@wakaleo Tried to reach out to you but reachme@.... seems to be not working

 This is the mail system at host eitzen.at.

I'm sorry to have to inform you that your message could not
be delivered to one or more recipients. It's attached below.

For further assistance, please send mail to postmaster.

If you do so, please include this problem report. You can
delete your own text from the attached returned message.

                   The mail system

[<reachme@wakaleo.com>](mailto:reachme@wakaleo.com): host aspmx.l.google.com[173.194.76.27] said: 550-5.1.1
    The email account that you tried to reach does not exist. Please try
    550-5.1.1 double-checking the recipient's email address for typos or
    550-5.1.1 unnecessary spaces. For more information, go to 550 5.1.1
    https://support.google.com/mail/?p=NoSuchUser
    5b1f17b1804b1-42bb6deacbdsi70759145e9.19 - gsmtp (in reply to RCPT TO
    command)

**Comment by wakaleo** [comment]
This should be good - I will take a look this evening

**Comment by eitzenbe** [comment]
sent message via l...in

**Comment by wakaleo** [comment]
There are some failing tests in WhenFormattingDataForTheHTMLReports.groovy and WhenFormattingForHTML.java - they need to be updated to reflect the new behaviour, or in some cases they may indicate regressions in other behaviour that need to be integrated in your code. You can run these tests in isolation 

e.g.
<img width="876" alt="image" src="https://github.com/user-attachments/assets/15deb18a-13b4-4f39-a617-767943aaa066">

**Comment by eitzenbe** [comment]
I will check tomorrow

04.09.2024 20:32:00 John Ferguson Smart ***@***.***>:

**Comment by eitzenbe** [comment]
@wakaleo Hmmm the method htmlCompatible in the formatter class should escape all special characters assuming it is a text that is put in or? If so the test case is wrong, without javadoc its hard to see whether the test is wrong or the new method impl I did. From the name "plainHtmlCompatible" I deducted that < and > characters should be escaped too as these could lead to problems in the title section of scenarios in the report.

The test method name implies you are actually testing the renderMarkdown method in which case I would just remove the chained call to htmlCompatible., What do you say?

br
Thomas

**Comment by wakaleo** [comment]
The test method names are usually a good indication of the indented behaviour. Sometimes we don't want to escape < and > because we actually do want to render the HTML as is (particularly if we are allowing markdown rendering such as is the case for test titles and descriptions). For example, this one looks like there is an issue rendering the markdown notation:
<img width="1515" alt="image" src="https://github.com/user-attachments/assets/3df40168-6594-42cf-a440-c0bd5ffc70df">

**Comment by eitzenbe** [comment]
reworkign the code to not touch the formatter classes at all, PR incoming ;)

**Comment by wakaleo** [comment]
> Ok so fixed it with staying inside ftl to avoid side effects, but WHY are you replacing "<" in example table values with "{" this may cause "havoc" if test data on purpose contains a "<"... Not going down that rabbit whole but might be a good idea to rework that at a later time ;)
> 
> ![image](https://private-user-images.githubusercontent.com/5560776/364742096-ff9b0f89-78f9-453e-8d81-a1848f5eb98e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU1MzM4MjIsIm5iZiI6MTcyNTUzMzUyMiwicGF0aCI6Ii81NTYwNzc2LzM2NDc0MjA5Ni1mZjliMGY4OS03OGY5LTQ1M2UtOGQ4MS1hMTg0OGY1ZWI5OGUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDVUMTA1MjAyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OWJmYjM4MTRiZTQ5M2MwNTU5YjZkZWI4NDRlNDE4MzA3YTU3NGUxNGNkNjQ1MzcwMTQ5OWVhZGQwZjk1YTJiOCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.YUyeq0xmHVuE3hID98CE1foyEORtpcsV-5hordHpFuM)

Thanks. This is a good point and there was probably a very good reason at the time which I don't recall now 🤷
